source/extensions/dynamic_modules/abi.h

```diff
@@ -230,6 +230,147 @@ typedef enum {
+  envoy_dynamic_module_type_attribute_id_ConnectionUriSanPeerCertificat,
+  // connection.sha256_peer_certificate_digest
+  envoy_dynamic_module_type_attribute_id_ConnectionSha256PeerCertificate,
+  // connection.transport_failure_reason
+  envoy_dynamic_module_type_attribute_id_ConnectionTransportFailureReaso,
+  // connection.termination_details
+  envoy_dynamic_module_type_attribute_id_ConnectionTerminationDetails,
+  // upstream.address
+  envoy_dynamic_module_type_attribute_id_UpstreamAddress,
+  // upstream.port
+  envoy_dynamic_module_type_attribute_id_UpstreamPort,
+  // upstream.tls_version
+  envoy_dynamic_module_type_attribute_id_UpstreamTlsVersion,
+  // upstream.subject_local_certificate
+  envoy_dynamic_module_type_attribute_id_UpstreamSubjectLocalCertificate,
+  // upstream.subject_peer_certificate
+  envoy_dynamic_module_type_attribute_id_UpstreamSubjectPeerCertificate,
+  // upstream.dns_san_local_certificate
+  envoy_dynamic_module_type_attribute_id_UpstreamDnsSanLocalCertificate,
+  // upstream.dns_san_peer_certificate
+  envoy_dynamic_module_type_attribute_id_UpstreamDnsSanPeerCertificate,
+  // upstream.uri_san_local_certificate
+  envoy_dynamic_module_type_attribute_id_UpstreamUriSanLocalCertificate,
+  // upstream.uri_san_peer_certificate
+  envoy_dynamic_module_type_attribute_id_UpstreamUriSanPeerCertificate,
+  // upstream.sha256_peer_certificate_digest
+  envoy_dynamic_module_type_attribute_id_UpstreamSha256PeerCertificateD,
+  // upstream.local_address
+  envoy_dynamic_module_type_attribute_id_UpstreamLocalAddress,
+  // upstream.transport_failure_reason
+  envoy_dynamic_module_type_attribute_id_UpstreamTransportFailureReason,
+  // upstream.request_attempt_count
+  envoy_dynamic_module_type_attribute_id_UpstreamRequestAttemptCount,
+  // upstream.cx_pool_ready_duration
+  envoy_dynamic_module_type_attribute_id_UpstreamCxPoolReadyDuration,
+  // upstream.locality
+  envoy_dynamic_module_type_attribute_id_UpstreamLocality,
+  // xds.node
+  envoy_dynamic_module_type_attribute_id_XdsNode,
+  // xds.cluster_name
+  envoy_dynamic_module_type_attribute_id_XdsClusterName,
+  // xds.cluster_metadata
+  envoy_dynamic_module_type_attribute_id_XdsClusterMetadata,
+  // xds.listener_direction
+  envoy_dynamic_module_type_attribute_id_XdsListenerDirection,
+  // xds.listener_metadata
+  envoy_dynamic_module_type_attribute_id_XdsListenerMetadata,
+  // xds.route_name
+  envoy_dynamic_module_type_attribute_id_XdsRouteName,
+  // xds.route_metadata
+  envoy_dynamic_module_type_attribute_id_XdsRouteMetadata,
+  // xds.virtual_host_name
+  envoy_dynamic_module_type_attribute_id_XdsVirtualHostName,
+  // xds.virtual_host_metadata
+  envoy_dynamic_module_type_attribute_id_XdsVirtualHostMetadata,
+  // xds.upstream_host_metadata
+  envoy_dynamic_module_type_attribute_id_XdsUpstreamHostMetadata,
+  // xds.filter_chain_name
+  envoy_dynamic_module_type_attribute_id_XdsFilterChainName,
+} envoy_dynamic_module_type_attribute_id;
```

Comment:
note: the choice of enum vs accepting an opaque "path" is intentional as we've experienced in Wasm that paths resolution is unnecessarily complex and slow look up. IIRC Wasm is planning to migrate this enum style from the opaque "get_property([]string)" style.